File: HackerRank/World Codesprint 6/bonetrousle.cpp
#include <bits/stdc++.h>

#define lli long long int

using namespace std;

int main()
{
	int t;
	cin>>t;

	while(t--)
	{
		lli n,k,b;
		cin>>n>>k>>b;

		lli mxb = (b*(b+1))/2;
		lli mxk = (k*(k+1))/2;
		if(n >= mxb && n <= mxk)
		{
			lli A[b];
			for(lli i=0;i<b;i++)
			{
				A[i] = i+1;
				n -= (i+1);
			}

			lli i = b-1;
			while(n && i>=0)
			{
				lli d = min(n,abs(A[i]-k));

				A[i] += d;

				k--;
				n -= d;
				i--;
			}

			if(n)
				printf("-1\n");
			else
			{
				for(int i=0;i<b;i++)
					printf("%lld ",A[i]);
				printf("\n");
			}
		}
		else
			printf("-1\n");
	}
}

File: HackerRank/World Codesprint 6/bonetrousle.py
t = int(input())

for tst in range(t):
	s = input()
	s = s.split()
	n = int(s[0])
	k = int(s[1])
	b = int(s[2])

	mxb = (b*(b+1))//2;
	mxk = (k*(k+1))//2;

	if(n >= mxb and n <= mxk):
		A = [i+1 for i in range(b)]
		n -= mxb

		i = b-1
		while(n != 0 and i>=0):
			d = min(n,abs(A[i]-k))

			A[i] += d
			k -= 1
			n -= d
			i -= 1

		if n != 0:
			print("-1")
		else:
			s = str(A)
			s = s.replace(",","")
			s = s.replace("[","")
			s = s.replace("]","")
			print(s)
	else:
		print("-1")